internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -289,3 +310,63 @@ func setCPUQuota(cpuMountPoint, parentDir string, c *oci.Container, enable bool)
 
 	return nil
 }
+
+func setRPSLoadBalancing(c *oci.Container, sysDevicesDir, onlineCPUsSysFile string, enable bool) error {
+	lspec := c.Spec().Linux
+	if lspec == nil ||
+		lspec.Resources == nil ||
+		lspec.Resources.CPU == nil ||
+		lspec.Resources.CPU.Cpus == "" {
+		return errors.Errorf("cannot find container %s CPUs", c.ID())
+	}
+
+	content, err := ioutil.ReadFile(onlineCPUsSysFile)
+	if err != nil {
+		return err
+	}
+	cpus := strings.Trim(string(content), "\n")
+
+	machineCPUs, err := cpuset.Parse(cpus)
+	if err != nil {
+		return err
+	}
+
+	machineCPUsMask, _, err := computeCPUmask(cpus, "", true)
+	if err != nil {
+		return err
+	}
+
+	return filepath.Walk(sysDevicesDir, func(path string, info os.FileInfo, err error) error {
```

Comment:
can we narrow down this path a bit further? for example: can we scan only the `/sys/class/net` tree? not really sure this is the best subtree, just an example.

Reply:
Sadly we can't. These files ar e only under /sys/device and under different sub-directories.
However only the net devices have the rps_cpus file, so we should be fine.
`
/sys/devices/pci0000:00/0000:00:03.0/virtio0/net/ens3/queues/rx-0/rps_cpus
/sys/devices/virtual/net/vxlan_sys_4789/queues/rx-0/rps_cpus
/sys/devices/virtual/net/veth2fdfb5e3/queues/rx-0/rps_cpus
/sys/devices/virtual/net/tun0/queues/rx-0/rps_cpus
/sys/devices/virtual/net/vetha8646254/queues/rx-0/rps_cpus
/sys/devices/virtual/net/vethf2fa53ed/queues/rx-0/rps_cpus
/sys/devices/virtual/net/br0/queues/rx-0/rps_cpus
/sys/devices/virtual/net/ovs-system/queues/rx-0/rps_cpus
/sys/devices/virtual/net/veth67ff0295/queues/rx-0/rps_cpus
/sys/devices/virtual/net/veth6afda312/queues/rx-0/rps_cpus
/sys/devices/virtual/net/veth84c456c3/queues/rx-0/rps_cpus
/sys/devices/virtual/net/lo/queues/rx-0/rps_cpus
/sys/devices/virtual/net/vethcd6422cb/queues/rx-0/rps_cpus
/sys/devices/virtual/net/veth321c2a31/queues/rx-0/rps_cpus
/sys/devices/virtual/net/veth55d926ae/queues/rx-0/rps_cpus
`